src/Pipelines/Stages/ReIndexStage.php

```diff
@@ -56,15 +59,48 @@ public function __invoke($payload)
                 ],
             ]);
 
-            do {
+            // Use a progress bar to indicate how far a long the re-indexing has come
+            $this->renderProgressBar($task);
+        }
+
+        return $payload;
+    }
+
+    /**
+     * Renders a progress bar until the specified re-indexing task is completed
+     *
+     * @param array $task
+     */
+    protected function renderProgressBar(array $task): void
+    {
+        // Use a progress bar to indicate how far a long the re-indexing has come
+        $progressBar = null;
+
+        do {
+            $response = ['completed' => false];
+
+            // Ignore ServerErrorResponseException, re-indexing can make these requests time out
```

Comment:
Is there a risk for an endless loop?
Are there any other cases than time out to throw `ServerErrorResponseException` ?
What if the Elastic server goes down during the loop?

Reply:
For this particular API endpoint there shouldn't be much that can trigger an exception except timeouts. If Elasticsearch completely dies it should eventually come back at some point and allow the loop to exit, so I think this is such an edge case that it's not worth handling.